### PR TITLE
madpack: Add madpack option to run unit tests.

### PR DESCRIPTION
**Pull request by njayaram2**
JIRA: MADLIB-1252

Unit tests in MADlib are written in python files, that are located in
the ...<module_name>/test/unit_tests/ folders, whose names begin with
the prefix "test_". This commit adds a new madpack option to run unit
tests similar to how we run install and dev checks.

- The new option added is: `unit-test`.
- Sample usage (on a postgres database, with MADlib installed on
  database `madlib`):
  * Run unit tests on all modules that have it defined:
      src/bin/madpack -p postgres -c /madlib unit-test
  * Run unit tests only for the `convex` module:
      src/bin/madpack -p postgres -c /madlib unit-test -t convex
  * Run unit tests only for the `convex` and decision trees module:
      src/bin/madpack -p postgres -c /madlib unit-test -t
      convex,recursive_partitioning/decision_tree
- Add command to run all unit tests in Jenkins build script.

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/544/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/546/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/547/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/548/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/549/

File: src/ports/postgres/modules/convex/test/unit_tests/test_mlp_igd.py_in [hunk 1]
```diff
@@ -126,10 +126,11 @@ class MLPMiniBatchTestCase(unittest.TestCase):
                                                 'dependent_varname': 'value',
                                                 'class_values': 'regression',
                                                 'grouping_cols': 'value',
+                                                'dependent_vartype': 'value',
                                                 'foo': 'bar'}]
         self.module = self.subject.MLPMinibatchPreProcessor("input")
         self.assertTrue(self.module.preprocessed_summary_dict)
-        self.assertEqual(5, len(self.module.preprocessed_summary_dict))
+        self.assertEqual(6, len(self.module.preprocessed_summary_dict))
```
Comment (iyerr3) [inline src/ports/postgres/modules/convex/test/unit_tests/test_mlp_igd.py_in]: Let's put this as a separate commit (can be pushed to master already) since it's unrelated to this PR.

File: src/madpack/madpack.py [hunk 2]
```diff
@@ -809,22 +811,100 @@ def parse_arguments():
...
+def run_unit_tests(args, testcase):
+    """
+        Run unit tests.
+    """
+    if not _is_madlib_installation_valid_for_tests(args['schema'],
+                                                   args['db_madlib_ver']):
+        return
+    modset = _get_modset_for_tests(testcase, 'test_')
+    # Loop through all modules and run unit tests
+    for moduleinfo in portspecs['modules']:
```
Comment (iyerr3) [inline src/madpack/madpack.py]: We've a similar loop/pattern in multiple places within `madpack.py`. Can we refactor them all to simplify these functions.

**Comment by njayaram2** [comment]
Thank you for the comments @iyerr3 , will do the needful.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/555/
